tcp/stream_factory.go

```diff
@@ -122,18 +126,29 @@ func (f *streamFactory) New(net, transport gopacket.Flow) tcpassembly.Stream {
 		}),
 	}
 
-	f.out <- r
+	f.outMu.Lock()
+	if !f.closed {
+		f.out <- r
+	}
+	f.outMu.Unlock()
```

Comment:
Should this be handled earlier in New?
```
func New() {
f.outMu.Lock()
defer f.outMu.Unlock()
if f.closed {
 return nil   ( or whatever makes sense here when the factory is closed )
}
...
f.out <- r
```
Stinks we'd hold the lock longer for each call, but it prevents us from doing a bunch of setup that is otherwise unnecessary.

Reply:
The `closed` state is only true when vhs is shutting down so I don't want to introduce more lock contention while vhs is actually running. This bool is in place to keep us from writing to the `out` channel once it has been closed so it makes the most sense to leave it as-is.